gulp-tasks/build.js

```diff
@@ -1,37 +1,55 @@
-var gulp = require('gulp'),
-    minCSS = require('gulp-clean-css'),
-    jshint = require('gulp-jshint'),
-    stylish = require('jshint-stylish'),
-    rev = require('gulp-rev'),
-    uglify = require('gulp-uglify'),
-    del = require('del');
-    usemin = require('gulp-usemin')
+var gulp = require('gulp');
+var useref = require('gulp-useref');
+var gulpif = require('gulp-if');
+var cleanCss = require('gulp-clean-css');
+var replace = require('gulp-replace');
+var uglify = require('gulp-uglify');
+var htmlmin = require('gulp-htmlmin');
+var rev = require('gulp-rev');
+var revReplace = require('gulp-rev-replace');
+var del = require('del');
 
-//IMPORT FILES FROM DIST
-gulp.task('imagemin', function() {
-  return del(['dist/app/assets/images/*']), {force: true}, gulp.src('dev/app/assets/images/*')
-    .pipe(gulp.dest('dist/app/assets/images'))
+/**
+ * Copy static files from /dev to /dist
+ */
+
+gulp.task('copy:fonts', function () {
+  del('dist/app/assets/fonts/**/*.{ttf,wof,eof,svg}');
+  return gulp.src('./bower_components/font-awesome/fonts/**/*.{ttf,woff,eof,svg}*')
+    .pipe(gulp.dest('dist/app/assets/fonts/'));
 });
 
-gulp.task('data', function () {
-  return del(['dist/app/data/*']), {force: true}, gulp.src('dev/app/data/*.json')
-    .pipe(gulp.dest('dist/app/data/'));
+gulp.task('copy:images', function () {
+  del('dist/app/assets/images/**/*.*');
+  return gulp.src('dev/app/assets/images/**/*.*')
+    .pipe(gulp.dest('dist/app/assets/images'));
 });
 
- gulp.task('copyfonts', function() {
-   return del(['dist/app/assets/fonts/*']), {force: true}, gulp.src('./bower_components/font-awesome/fonts/**/*.{ttf,woff,eof,svg}*')
-    .pipe(gulp.dest('dist/app/assets/fonts/'));
- });
+gulp.task('copy:data', function () {
+  del('dist/app/data/*');
+  return gulp.src('dev/app/data/*.json')
+    .pipe(gulp.dest('dist/app/data/'));
+});
 
-//BUNDLE FILES INTO ONE
-gulp.task('usemin', function () {
-  return gulp.src('dev/app/**/*.html')
-      .pipe(usemin({
-        js: [rev()],
-        css:[minCSS(),rev()]
-              }))
-      .pipe(gulp.dest('dist/app'));
+/**
```

Comment:
ok - send me the links. I reckon we need to split task on those .

I reckon is this one http://browserify.org/
I think it s more urgent than the rest. We ll need to divide tasks.